app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -331,10 +331,11 @@ def save_switches_inventory(host, hashes)
     host.switches.delete(deletes)
   end
 
-  def remove_obsolete_switches
+  def remove_obsolete_switches(switches)
     # delete from switches as s where s.shared is NULL and s.id not in (select switch_id from host_switches)
     # delete from switches as s where s.shared = 't' and s.id not in (select switch_id from host_switches)
-    Switch.where.not(:id => HostSwitch.all.collect(&:switch).uniq).destroy_all
+    switch_ids = HostSwitch.all.collect(&:switch_id)
+    switches.where.not(:id => switch_ids).destroy_all
```

Comment:
@agrare It's works. Is it ok for you?

Reply:
At [this point](https://github.com/ManageIQ/manageiq/pull/17390/files#diff-50f9feed046a3410ac5531492059a6fbR81) the switches have already been saved so this won't actually disconnect anything.

I suggested collecting the IDs [here](https://github.com/CharlleDaniel/manageiq/blob/f880ad7288241343fb8e31ef078c8e05db3690e7/app/models/ems_refresh/save_inventory_infra.rb#L67) since that will be the original list.  You can then destroy the `old_switch_ids - switch_ids`